rs/index/src/ivf/files/invalidated_ids.rs

```diff
@@ -142,6 +140,43 @@ impl InvalidatedIdsStorage {
         Ok(())
     }
 
+    /// Batch invalidates multiple `InvalidatedUserDocId`s without fsyncing on every invalidation.
+    /// Performs fsync at the end of each file.
+    pub fn invalidate_batch(&mut self, invalidations: &[InvalidatedUserDocId]) -> Result<()> {
+        let mut buffer = Vec::with_capacity(BYTES_PER_INVALIDATION);
+
+        for invalidation in invalidations {
+            // Check if the current file has enough space for this invalidation
+            if self.current_offset == self.backing_file_size {
+                if !buffer.is_empty() {
+                    let file = &mut self.files[self.current_backing_id as usize];
+                    file.write_all(&buffer)?;
+                    file.sync_all()?;
+
+                    // Clear the buffer and create a new backing file
+                    buffer.clear();
+                }
+                self.new_backing_file()?;
+            }
+
+            // Write user_id and doc_id into the buffer
+            buffer.extend_from_slice(&invalidation.user_id.to_le_bytes());
+            buffer.extend_from_slice(&invalidation.doc_id.to_le_bytes());
```

Comment:
does this buffer keep growing? let's preallocate & cap the size, and once the buffer is large enough, just write to file, and reset the buffer (no need to sync. just sync either at the end or when you create a new file)

Reply:
It's preallocated with the size of 1 invalidation now. I guess the problem with preallocating more space is it may exceed the remaining space of current backing file, which makes the logic to handle everything more complicated.

Reply:
and not it does not keep growing, we clear it up when allocating new backing file